To: vim_dev@googlegroups.com
Subject: Patch 7.3.174
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.174
Problem:    When Exuberant ctags binary is exctags it's not found.
Solution:   Add configure check for exctags. (Hong Xu)
Files:	    src/configure.in, src/auto/configure


*** ../vim-7.3.173/src/configure.in	2011-02-09 17:42:53.000000000 +0100
--- src/configure.in	2011-05-05 17:18:21.000000000 +0200
***************
*** 2619,2625 ****
    dnl  On HP-UX 10.10 termcap or termlib should be used instead of
    dnl  curses, because curses is much slower.
    dnl  Newer versions of ncurses are preferred over anything, except
!   dnl  when tinfo has been split off, it conains all we need.
    dnl  Older versions of ncurses have bugs, get a new one!
    dnl  Digital Unix (OSF1) should use curses (Ronald Schild).
    dnl  On SCO Openserver should prefer termlib (Roger Cornelius).
--- 2619,2625 ----
    dnl  On HP-UX 10.10 termcap or termlib should be used instead of
    dnl  curses, because curses is much slower.
    dnl  Newer versions of ncurses are preferred over anything, except
!   dnl  when tinfo has been split off, it contains all we need.
    dnl  Older versions of ncurses have bugs, get a new one!
    dnl  Digital Unix (OSF1) should use curses (Ronald Schild).
    dnl  On SCO Openserver should prefer termlib (Roger Cornelius).
***************
*** 3370,3376 ****
  AC_CHECK_LIB(xpg4, _xpg4_setrunelocale, [LIBS="$LIBS -lxpg4"],,)
  
  dnl Check how we can run ctags.  Default to "ctags" when nothing works.
! dnl --version for Exuberant ctags (preferred)
  dnl       Add --fields=+S to get function signatures for omni completion.
  dnl -t for typedefs (many ctags have this)
  dnl -s for static functions (Elvis ctags only?)
--- 3370,3376 ----
  AC_CHECK_LIB(xpg4, _xpg4_setrunelocale, [LIBS="$LIBS -lxpg4"],,)
  
  dnl Check how we can run ctags.  Default to "ctags" when nothing works.
! dnl Use --version to detect Exuberant ctags (preferred)
  dnl       Add --fields=+S to get function signatures for omni completion.
  dnl -t for typedefs (many ctags have this)
  dnl -s for static functions (Elvis ctags only?)
***************
*** 3378,3384 ****
  dnl -i+m to test for older Exuberant ctags
  AC_MSG_CHECKING(how to create tags)
  test -f tags && mv tags tags.save
! if (eval ctags --version /dev/null | grep Exuberant) < /dev/null 1>&AC_FD_CC 2>&1; then
    TAGPRG="ctags -I INIT+ --fields=+S"
  else
    TAGPRG="ctags"
--- 3378,3386 ----
  dnl -i+m to test for older Exuberant ctags
  AC_MSG_CHECKING(how to create tags)
  test -f tags && mv tags tags.save
! if (eval exctags --version /dev/null | grep Exuberant) < /dev/null 1>&AC_FD_CC 2>&1; then
!   TAGPRG="exctags -I INIT+ --fields=+S"
! elif (eval ctags --version /dev/null | grep Exuberant) < /dev/null 1>&AC_FD_CC 2>&1; then
    TAGPRG="ctags -I INIT+ --fields=+S"
  else
    TAGPRG="ctags"
*** ../vim-7.3.173/src/auto/configure	2011-02-09 17:42:53.000000000 +0100
--- src/auto/configure	2011-05-05 17:19:26.000000000 +0200
***************
*** 12031,12037 ****
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking how to create tags" >&5
  $as_echo_n "checking how to create tags... " >&6; }
  test -f tags && mv tags tags.save
! if (eval ctags --version /dev/null | grep Exuberant) < /dev/null 1>&5 2>&1; then
    TAGPRG="ctags -I INIT+ --fields=+S"
  else
    TAGPRG="ctags"
--- 12031,12039 ----
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking how to create tags" >&5
  $as_echo_n "checking how to create tags... " >&6; }
  test -f tags && mv tags tags.save
! if (eval exctags --version /dev/null | grep Exuberant) < /dev/null 1>&5 2>&1; then
!   TAGPRG="exctags -I INIT+ --fields=+S"
! elif (eval ctags --version /dev/null | grep Exuberant) < /dev/null 1>&5 2>&1; then
    TAGPRG="ctags -I INIT+ --fields=+S"
  else
    TAGPRG="ctags"
*** ../vim-7.3.173/src/version.c	2011-05-05 17:14:07.000000000 +0200
--- src/version.c	2011-05-05 17:19:37.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     174,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
54. You start tilting your head sideways to smile. :-)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
